[IsDeprecatedWeakRefSmartPointerException] Make WebLockRegistryProxy ref-counted

RupinMittal opened the pull request:
#### c7f286cac5e719c0493d44658f8a381f0e8b6738
<pre>
[IsDeprecatedWeakRefSmartPointerException] Make WebLockRegistryProxy ref-counted
<a href="https://bugs.webkit.org/show_bug.cgi?id=281465">https://bugs.webkit.org/show_bug.cgi?id=281465</a>
<a href="https://rdar.apple.com/137916057">rdar://137916057</a>

Reviewed by Chris Dumez.

Forward WebLockRegistryProxy&apos;s ref/deref to its owner WebProcessProxy.

* Source/WebKit/UIProcess/WebLockRegistryProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setWebsiteDataStore):
(WebKit::WebProcessProxy::shutDown):

Canonical link: <a href="https://commits.webkit.org/285274@main">https://commits.webkit.org/285274@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80b34823b1da7338370fec56a68d35b5a5cccabb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71993 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51413 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24778 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76153 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23202 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59214 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23022 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56797 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15304 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75060 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46620 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62016 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37240 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43283 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21548 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65189 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19851 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77833 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16233 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19026 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65285 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16279 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62040 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/64535 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12724 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6366 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11059 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47211 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/1995 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48280 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49567 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48024 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

RupinMittal commented on Source/WebKit/UIProcess/WebLockRegistryProxy.cpp:
```diff
@@ -47,17 +47,21 @@ WTF_MAKE_TZONE_ALLOCATED_IMPL(WebLockRegistryProxy);
 WebLockRegistryProxy::WebLockRegistryProxy(WebProcessProxy& process)
```
At the top of this file, there are two `#define` statements (the message checks) that use `m_process`. Do we need to null check `m_process` inside of these message checks?

cdumez replied:
Non-issue if you apply my other review comment.

cdumez commented on Source/WebKit/UIProcess/WebLockRegistryProxy.h:
```diff
@@ -30,27 +30,26 @@
 struct SharedPreferencesForWebProcess;
 
-class WebLockRegistryProxy final : public IPC::MessageReceiver {
+class WebLockRegistryProxy final : public IPC::MessageReceiver, public RefCounted<WebLockRegistryProxy> {
```
It would be better to forward to ref-counting to the owner WebProcessProxy.

RupinMittal replied:
Fixed!

cdumez commented on Source/WebKit/UIProcess/WebLockRegistryProxy.h:
```diff
@@ -69,7 +66,7 @@ class WebLockRegistryProxy final : public IPC::MessageReceiver {
     Ref<WebProcessProxy> protectedProcess() const { return m_process.get(); }
 
-    CheckedRef<WebProcessProxy> m_process;
+    WeakRef<WebProcessProxy> m_process;
```
I don't think we need this change?

RupinMittal replied:
Fixed!

cdumez reviewed (approved):
LGTM

RupinMittal committed:
[IsDeprecatedWeakRefSmartPointerException] Make WebLockRegistryProxy ref-counted
https://bugs.webkit.org/show_bug.cgi?id=281465
rdar://137916057

Reviewed by Chris Dumez.

Forward WebLockRegistryProxy's ref/deref to its owner WebProcessProxy.

* Source/WebKit/UIProcess/WebLockRegistryProxy.h:
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::setWebsiteDataStore):
(WebKit::WebProcessProxy::shutDown):

Canonical link: https://commits.webkit.org/285274@main
